src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ReferenceEqualityComparer.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Collections.Generic
+{
+    /// <summary>
+    /// An <see cref="IEqualityComparer{Object}"/> that uses reference equality (<see cref="object.ReferenceEquals(object?, object?)"/>)
+    /// instead of value equality (<see cref="object.Equals(object?)"/>) when comparing two object instances.
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="ReferenceEqualityComparer"/> type cannot be instantiated. Instead, use the <see cref="Instance"/> property
+    /// to access the singleton instance of this type.
+    /// </remarks>
+    public sealed class ReferenceEqualityComparer : IEqualityComparer<object>, IEqualityComparer
+    {
+        private ReferenceEqualityComparer() { }
+
+        /// <summary>
+        /// Gets the singleton <see cref="ReferenceEqualityComparer"/> instance.
+        /// </summary>
+        public static ReferenceEqualityComparer Instance { get; } = new ReferenceEqualityComparer();
```

Comment:
This is going to create a new instance every time is called, wouldn't be better to use a Field instead to avoid the allocation at every time something that uses `Instance` is being executed?

That is the reason why I opted-in for a Field on [this implementation](https://github.com/dotnet/runtime/blob/19d2d5c2148b3271e3f88d323dfee3a3c9f16bd0/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReferenceEqualsEqualityComparer.cs#L15).

Reply:
The syntax `=> new T()` would be a new instance every time. The syntax `{ get; } = new T()` is a singleton pattern. This is also checked in the below unit test.

https://github.com/dotnet/runtime/blob/16ba551952269e4395423d54e7a464be321f9295/src/libraries/System.Collections/tests/Generic/Comparers/ReferenceEqualityComparer.Tests.cs#L20-L29

Reply:
Oh I didn't know that. Cool.